### ISSUE TITLE
Broken CG calculation with 3-stage model

### ISSUE THREAD

**Issue by neilweinstock** [openrocket/openrocket#755]
On the attached model, CG calculation breaks when third stage is added.
[Talos Terrier Recruit.ork.zip](https://github.com/openrocket/openrocket/files/5109921/Talos.Terrier.Recruit.ork.zip)

Here it is with the top stage ("Recruit") only.  The CG matches the override set for the stage (15"):
![image](https://user-images.githubusercontent.com/243872/90911978-cb6a3400-e3a7-11ea-884f-6143976ce8b5.png)

Here it is with the next stage (Terrier) included.  CG moves back to a reasonable place:
![image](https://user-images.githubusercontent.com/243872/90912059-efc61080-e3a7-11ea-9cf5-467e738eba54.png)

Now we add the bottom stage (Talos).  For some reason, the CG moves *forward*:
![image](https://user-images.githubusercontent.com/243872/90912115-09ffee80-e3a8-11ea-8f7e-c878c66bf1aa.png)

The problem seems to be caused by the CG override of the "Talos" stage.  If that is disabled, CG moves back to a more reasonable position:
![image](https://user-images.githubusercontent.com/243872/90912261-4d5a5d00-e3a8-11ea-8754-a20d3dd4d932.png)

**Comment by teyrana** [comment]
@hcraigmiller 
Thank you for doing all the work to track down the root cause here!  That makes the bug fix much quicker in turning around :)

**Comment by neilweinstock** [comment]
@hcraigmiller 
Yes indeed, very nice work.  Option 2 is the only one that makes any sense; CG override of each stage should be relative to the front of that stage.

**Comment by hcraigmiller** [comment]
I found another quirk with the CG calculation; not sure if this is addressed in the work done by **teyrana**; if so, disregard.

The design file is attached:
[Staged Tubes.Analysis.zip](https://github.com/openrocket/openrocket/files/5159415/Staged.Tubes.Analysis.zip)

Start by setting all component mass overrides to 1 and CG overrides to 4, then set all stage mass overrides to 1 and CG overrides to 4. Then turn off all CG overrides, both component and stage. **You cannot save these settings and reopen the file (all unchecked settings are set to 0 when reopened)**. 

When you turn the stages on and off, the result is what you would expect:
![Tubes Component Mass Override-1 Stage Mass Override-1 CG Override-N](https://user-images.githubusercontent.com/68821492/91920994-4290a980-ec7f-11ea-9549-4aeee953586b.png)
And, this is the same result you get if all of the body tube overrides are turned off.

But, if you set the body tube overrides to 0, even though the stage overrides should be supplanting the body tube overrides, you get an entirely different outcome:
![Tubes Component Mass Override-0 Stage Mass Override-1 Stage CG Override-N](https://user-images.githubusercontent.com/68821492/91920597-6dc6c900-ec7e-11ea-8624-ccf71e7efdaf.png)
But this time, as above, if you turn off all of the body tube overrides, everything is as above, what you would expect.

In so far as I have determined at this point, problems only occur if a body tube override is set to zero, if set even to .001, no problems occur. The problem does not appear to occur with inner tubes either. Splitting the Stage 1 tube, then comparing CG (1) without overriding the body tube masses, (2) overriding both body tube masses with 1, (3) overriding the top body tube mass with 0 and the bottom body tube mass with 1, (4) overriding the top body tube mass with 1 and the bottom body tube mass with 0, (5) overriding both body tube masses with 0, results in:
![Tubes Component Mass Override-0-1 Stage Mass Override-1 CG Override-N](https://user-images.githubusercontent.com/68821492/91922957-11ff3e80-ec84-11ea-9893-ba0527242d43.png)

Results 3 and 4 are the same as you would get if you used .001 instead of zero, but that does not explain the result in 5 (if you override both tubes to a mass of .001, the CG is where the tunes meet, as would be expected.

It appears that the CG calculation using stage mass overrides does not entirely supplant overridden component masses that are set to zero. 

Suggestion:
1.  A stage mass override should entirely supplant component masses within that stage, whether overridden or not, whether set at zero or not; OR
2.  Use "NA" for both display and numerical output when the mass of an overridden primary component (body tubes at this point) is set to zero, and give some warning as to why.
